src/solvers/dgsem_tree/indicators_2d.jl

```diff
@@ -185,6 +185,100 @@ function (löhner::IndicatorLöhner)(u::AbstractArray{<:Any,4},
 end
 
 
+# this method is used when the indicator is constructed as for shock-capturing volume integrals
+function create_cache(indicator::Type{IndicatorIDP}, equations::AbstractEquations{2}, basis::LobattoLegendreBasis, number_bounds)
+  ContainerShockCapturingIndicator = Trixi.ContainerShockCapturingIndicatorIDP{real(basis)}(0, nnodes(basis), number_bounds)
+
+  cache = (; ContainerShockCapturingIndicator)
+
+  return cache
+end
+
+function (indicator_IDP::IndicatorIDP)(u::AbstractArray{<:Any,4}, semi, dg::DGSEM, t, dt; kwargs...)
+  @unpack alpha = indicator_IDP.cache.ContainerShockCapturingIndicator
+  alpha .= 0.0
+
+  indicator_IDP.IDPPositivity  &&
+    @trixi_timeit timer() "IDPPositivity"  IDP_positivity!(alpha, indicator_IDP, u, dt, semi)
+
+  # Calculate alpha1 and alpha2
+  @unpack alpha1, alpha2 = indicator_IDP.cache.ContainerShockCapturingIndicator
+  @threaded for element in eachelement(dg, semi.cache)
+    for j in eachnode(dg), i in 2:nnodes(dg)
+      alpha1[i, j, element] = max(alpha[i-1, j, element], alpha[i, j, element])
+    end
+    for j in 2:nnodes(dg), i in eachnode(dg)
+      alpha2[i, j, element] = max(alpha[i, j-1, element], alpha[i, j, element])
+    end
+    alpha1[1,            :, element] .= zero(eltype(alpha1))
+    alpha1[nnodes(dg)+1, :, element] .= zero(eltype(alpha1))
+    alpha2[:,            1, element] .= zero(eltype(alpha2))
+    alpha2[:, nnodes(dg)+1, element] .= zero(eltype(alpha2))
+  end
+
+  return nothing
+end
+
+@inline function IDP_positivity!(alpha, indicator_IDP, u, dt, semi)
```

Comment:
```suggestion
@inline function idp_positivity!(alpha, indicator, u, dt, semi)
```
`snake_case` and avoid redundancy